internal/server/statvar/fetcher/svg_fetcher.go

```diff
@@ -100,9 +102,56 @@ func FetchAllSVG(
 		}
 	}
 	if store.SQLClient != nil {
-		// Query for all the stat var group node
-		query :=
-			`
+		sqlResult, err := fetchSQLSVGs(store.SQLClient)
+		if err != nil {
+			return nil, err
+		}
+		for svgId, svgNode := range sqlResult {
+			if _, ok := result[svgId]; !ok {
+				result[svgId] = svgNode
+			} else {
+				hierarchy.MergeSVGNodes(result[svgId], svgNode)
+			}
+		}
+	}
+	// Recount all descendent stat vars after merging
+	hierarchy.AdjustDescendentSVCount(result, hierarchy.SvgRoot)
+	return result, nil
+}
+
+// Fetches SVGs from SQL.
+// First attempts to get it from sql cache and falls back to querying sql table.
+func fetchSQLSVGs(sqlClient *sql.DB) (map[string]*pb.StatVarGroupNode, error) {
+	// Try cache first.
+	svgCache, err := fetchSQLCacheSVGs(sqlClient)
+	if err != nil {
+		return map[string]*pb.StatVarGroupNode{}, err
+	}
+	if svgCache != nil {
+		// Cached data found => return it.
+		return svgCache.StatVarGroups, nil
+	}
+
+	// Query sql table.
```

Comment:
Add a TOOD to deprecate fetching from table?

Reply:
Done. Added TODO to the method definition.